lib/adapter.js

```diff
@@ -0,0 +1,497 @@
+  httpMetrics.count = 0;
+
+  return httpEvent;
+}
+
+function getProbeData(probeName) {
+
+  // Get data for given probe
+  var probeData = probeMetrics[probeName];
+  var count = probeData.count;
+  var average = count ? probeData.runningTotal / count : 0;
+
+  var probeEvent = {
+    count: count,
+    minimum: probeData.min,
+    maximum: probeData.max,
+    average: average
+  };
+
+  // Reset values
+  probeMetrics[probeName] = undefined;
+
+  return probeEvent;
+}
+
+function getHeapData() {
+  var jsonToReturn = heapMetrics;
+
+  // Reset JSON
+  heapMetrics = {};
+
+  return jsonToReturn;
+}
+
+function getGCHeapData() {
+  if (gcMetrics.count > 0) {
+    var usedHeap = gcMetrics.used / gcMetrics.count;
+
+    // Reset values
+    gcMetrics.used = 0;
+    gcMetrics.count = 0;
+  } else {
+    usedHeap = -1;
+  }
+  return usedHeap;
+}
+
+function getMessageData() {
+  var oldMessageMetrics = messageMetrics;
+  messageMetrics.sent = 0;
+  messageMetrics.received = 0;
+  return oldMessageMetrics;
+}
+
+function compareHistograms(oldHistogram, newHistogram) {
+  var objectCounts = {};
+
+  // Add all the instances currently in the heap.
+  for (var type in newHistogram) {
+    var newTuple = newHistogram[type];
+    var oldTuple = oldHistogram[type];
+    var diffTuple;
+    if (!oldTuple) {
+      diffTuple = newTuple;
+    } else {
+      diffTuple = {
+        count: newTuple.count - oldTuple.count,
+        size: newTuple.size - oldTuple.size,
+      };
+    }
+    objectCounts[type] = diffTuple;
+  }
+
+  // Account for any types where all instances have been deleted.
+  for (type in oldHistogram) {
+    oldTuple = oldHistogram[type];
+    diffTuple = objectCounts[type];
+    if (!diffTuple) {
+      diffTuple = {
+        count: 0 - oldTuple.count,
+        size: 0 - oldTuple.size,
+      };
+      objectCounts[type] = diffTuple;
+    }
+  }
+
+  return objectCounts;
+}
+
+function initialise() {
+  // Set up listeners for all the metrics here
+  createEventLoopListener();
+  createCpuListener();
+  createHeapListener();
+  createGCListener();
+  createMessageListener();
+  createHTTPListener();
+
+  for (var i in probeList) {
+    // Create listeners for all probes
+    createProbeListener(probeList[i]);
+  }
+}
+
+function updateMetrics(probeName, value) {
+
+  var metrics = probeMetrics[probeName];
+
+  // update metrics accordingly (and if they haven't been defined yet)
+  if (!metrics.min || value < metrics.min) {
+    metrics.min = value;
+  }
+  if (!metrics.max || value > metrics.max) {
+    metrics.max = value;
+  }
+
+  // Update other values accordingly
+  metrics.runningTotal += value;
+  metrics.count++;
+}
+
+// Export the desired functions
+var metrics = {
+  startCpuProfiling: startCpuProfiling,
+  stopCpuProfiling: stopCpuProfiling,
+  startTrackingObjects: startTrackingObjects,
+  stopTrackingObjects: stopTrackingObjects
+};
+
+var config = {
+  hostname: os.hostname(),
+  appName: path.basename(process.argv[1]),
```

Comment:
(for sam) look into appName and key, what they are (appName here will be sl-run.js, but that is not the name of the application being run)